osquery/tables/system/darwin/keychain_items.cpp

```diff
@@ -55,9 +55,17 @@ void genKeychainItem(const SecKeychainItemRef& item, QueryData& results) {
   // Any tag that does not exist for the item will prevent the entire result.
   for (const auto& attr_tag : kKeychainItemAttrs) {
     tags[0] = attr_tag.first;
-    SecKeychainItemCopyAttributesAndData(
+    auto status = SecKeychainItemCopyAttributesAndData(
         item, &info, &item_class, &attr_list, 0, nullptr);
 
+    if (status == errSecNoSuchAttr) {
+      // This attr does not exist, skip it.
+      continue;
+    } else if (status != errSecSuccess) {
+      // If this keychain item is not valid then don't add it to results.
```

Comment:
I think we will have to free `attr_list` here with `SecKeychainItemFreeAttributesAndData(...)`. 
Something along the lines of:

```cpp
if (attr_list != null) {
  SecKeychainItemFreeAttributesAndData(attr_list, nullptr);
}
```

Reply:
As far as I can tell we only get a non-NULL attr_list when the call succeeded (status == errSecSuccess) so I don't think we have anything to deallocate in the error case.